app/javascript/controllers/disable_controller.js

```diff
@@ -0,0 +1,39 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  connect() {
+    this.element.dataset.action = `${this.element.dataset.action || ""} submit->disable#disableForm`
+    this.uploadCount = 0
+  }
+
+  disableForm() {
+    this.submitButtons().forEach((el) => {
+      el.disabled = true
+      el.value = el.dataset.disableWith || el.value
+    })
+  }
+
+  uploading(disable = true) {
+    if (disable) {
+      this.uploadCount++
+    } else {
+      this.uploadCount--
+    }
+    this.submitButtons().forEach((el) => {
```

Comment:
is there a difference between when this loop runs and when `disableForm` runs?

Reply:
`disableForm` runs when the `<form>` itself submits.  So it disables the submit button, preventing users from double-clicking / double-saving.  But a save is already in-progress.  If you throw a `sleep 2` in the controller `#update`, you'll see it.

TMI: [Rails UJS](https://github.com/rails/rails-ujs) typically handles this in Rails apps.  But something appears incompatible/broken with Rails 7... and I really wanted the `Save` button to change to `Saving...` because I'm picky.